src/shared/typed-fetch.ts

```diff
@@ -0,0 +1,22 @@
+class FetchError extends Error {
```

Comment:
should we be exporting this type as well?

Reply:
I think its fine to keep this as internal code for now.